hopp/simulation/technologies/mhk_wave_source.py

```diff
@@ -0,0 +1,220 @@
+from typing import Any, Union, Optional
+import PySAM.MhkWave as MhkWave
+
+from hopp.simulation.technologies.power_source import PowerSource, SiteInfo, Sequence, logger
+from hopp.simulation.technologies.financial.custom_financial_model import CustomFinancialModel
+from hopp.simulation.technologies.financial.mhk_cost_model import MHKCosts
+#TODO: Add dispatch for Wave
+# hopp.dispatch.power_sources.wave_dispatch import WaveDispatch
+
+class MHKWavePlant(PowerSource):
+    _system_model: MhkWave.MhkWave
+    _financial_model: Union[Any, CustomFinancialModel]
+    # _layout:
+    # _dispatch:
+
+    def __init__(self,
+                 site: SiteInfo,
+                 mhk_config: dict,
+                 cost_model_inputs: Optional[dict] = None
+                 ):
+        """
+        Initialize MhkWavePlant class
+
+        Args:
+        `mhk_config` (dict): A dictionary containing MHK system configuration parameters.
+            Required keys:
```

Comment:
These input dicts are pretty involved and would benefit from being broken out into data classes, but we can do this later